### PR TITLE
get devops count by username

### PR DESCRIPTION
**Pull request by runzexia**
Signed-off-by: runzexia <runzexia@yunify.com>

**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
At present, the authentication of workspace and devops is not fully integrated, so there will be errors in some number of display pages. This scheme displays the number of users' own devopsproject as a workaround solution, and integrates in 3.0.

**Which issue(s) this PR fixes**:
fix gitlab-674

**Special notes for reviewers**:
/cc @zryfish @wansir

### CODE REVIEW

File: pkg/apis/network/v1alpha1/common.go [hunk 1]
```diff
@@ -1,8 +1,8 @@
 package v1alpha1
 
 import (
-	"kubesphere.io/kubesphere/pkg/apis/network/v1alpha1/numorstring"
 	corev1 "k8s.io/api/core/v1"
+	"kubesphere.io/kubesphere/pkg/apis/network/v1alpha1/numorstring"
```
Comment (zryfish) [inline pkg/apis/network/v1alpha1/common.go]: import path is ambiguous , why import network package into devops? consider move `numorstring` to a more public package, such as `utils`

File: pkg/apis/tenant/v1alpha2/register.go [hunk 2]
```diff
@@ -134,6 +134,14 @@ func addWebService(c *restful.Container) error {
 		Returns(http.StatusOK, ok, models.PageableResponse{}).
 		Doc("List the devops projects for the workspace member").
 		Metadata(restfulspec.KeyOpenAPITags, []string{constants.TenantResourcesTag}))
+	ws.Route(ws.GET("/devopscount").
+		To(tenant.GetDevOpsProjectsCount).
+		Param(ws.PathParameter("member", "workspace member's username")).
```
Comment (zryfish) [inline pkg/apis/tenant/v1alpha2/register.go]: there is no path parameter in the path

File: pkg/apiserver/tenant/tenant.go [hunk 3]
```diff
@@ -244,6 +244,19 @@ func ListDevopsProjects(req *restful.Request, resp *restful.Response) {
 	resp.WriteAsJson(result)
 }
 
+func GetDevOpsProjectsCount(req *restful.Request, resp *restful.Response) {
+	username := req.HeaderParameter(constants.UserNameHeader)
+
+	result, err := tenant.GetDevOpsProjectsCount(username)
+	if err != nil {
+		glog.Errorf("%+v", err)
```
Comment (zryfish) [inline pkg/apiserver/tenant/tenant.go]: use klog instead of glog

File: pkg/models/tenant/devops.go [hunk 4]
```diff
@@ -105,6 +106,36 @@ func ListDevopsProjects(workspace, username string, conditions *params.Condition
...
 }
 
+func GetDevOpsProjectsCount(username string) (uint32, error) {
+	dbconn := devops_mysql.OpenDatabase()
+
+	query := dbconn.Select(devops.GetColumnsFromStructWithPrefix(devops.DevOpsProjectTableName, devops.DevOpsProject{})...).
+		From(devops.DevOpsProjectTableName)
+	var sqconditions []dbr.Builder
+
+	switch username {
```
Comment (zryfish) [inline pkg/models/tenant/devops.go]: switch case in here is not clear, I prefer
```
if username != devops.KS_ADMIN
```

File: pkg/apis/network/v1alpha1/common.go (hunk 1 above)
Comment (runzexia) [reply pkg/apis/network/v1alpha1/common.go]: just gofmt here

**Pull request merged**